src/rdata/naptr.rs

```diff
@@ -0,0 +1,545 @@
+
+    fn try_flatten_into(self) -> Result<Naptr<TOcts, TName>, TOcts::Error> {
+        self.flatten()
+    }
+}
+
+//--- PartialEq and Eq
+
+impl<Octs, OtherOcts, Name, OtherName> PartialEq<Naptr<OtherOcts, OtherName>>
+    for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    OtherOcts: AsRef<[u8]>,
+    Name: ToName,
+    OtherName: ToName,
+{
+    fn eq(&self, other: &Naptr<OtherOcts, OtherName>) -> bool {
+        self.order == other.order
+            && self.preference == other.preference
+            && self.flags.eq(&other.flags)
+            && self.services.eq(&other.services)
+            && self.regexp.eq(&other.regexp)
+            && self.replacement.name_eq(&other.replacement)
+    }
+}
+
+impl<Octs: AsRef<[u8]>, Name: ToName> Eq for Naptr<Octs, Name> {}
+
+//--- PartialOrd, Ord, and CanonicalOrd
+
+impl<Octs, OtherOcts, Name, OtherName> PartialOrd<Naptr<OtherOcts, OtherName>>
+    for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    OtherOcts: AsRef<[u8]>,
+    Name: ToName,
+    OtherName: ToName,
+{
+    fn partial_cmp(
+        &self,
+        other: &Naptr<OtherOcts, OtherName>,
+    ) -> Option<Ordering> {
+        match self.order.partial_cmp(&other.order) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.preference.partial_cmp(&other.preference) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.flags.partial_cmp(&other.flags) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.services.partial_cmp(&other.services) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+        match self.regexp.partial_cmp(&other.regexp) {
+            Some(Ordering::Equal) => {}
+            other => return other,
+        }
+
+        Some(self.replacement.name_cmp(&other.replacement))
+    }
+}
+
+impl<Octs, OtherOcts, Name, OtherName>
+    CanonicalOrd<Naptr<OtherOcts, OtherName>> for Naptr<Octs, Name>
+where
+    Octs: AsRef<[u8]>,
+    OtherOcts: AsRef<[u8]>,
+    Name: ToName,
+    OtherName: ToName,
+{
+    fn canonical_cmp(&self, other: &Naptr<OtherOcts, OtherName>) -> Ordering {
+        match self.order.cmp(&other.order) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.preference.cmp(&other.preference) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.flags.canonical_cmp(&other.flags) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.services.canonical_cmp(&other.services) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+        match self.regexp.canonical_cmp(&other.regexp) {
+            Ordering::Equal => {}
+            other => return other,
+        }
+
+        self.replacement.name_cmp(&other.replacement)
```

Comment:
In other rdata implementations names in `impl CanonicalOrd` are compared using `lowercase_composed_cmp()` so I wonder if that should be the case here as well, e.g.

```suggestion
        self.replacement.lowercase_composed_cmp(&other.replacement)
```

Reply:
I think it should be case-insensitive in comparisons because it's a <domain-name>.